libpod/kube.go

```diff
@@ -325,36 +340,81 @@ func libpodMaxAndMinToResourceList(c *Container) (v1.ResourceList, v1.ResourceLi
 	return maxResources, minResources
 }
 
-func generateKubeVolumeMount(hostSourcePath string, mounts []specs.Mount) (v1.VolumeMount, error) {
+func generateKubeVolumeMount(hostSourcePath string, mounts []specs.Mount) (v1.VolumeMount, v1.Volume, error) {
```

Comment:
Might want to just pass the container in here - would be easier to work with. You loop through UserVolumes, identify named volume vs normal volume, and then add based on that. Should remove the need to deduplicate, for example.

Example code lives in https://github.com/containers/libpod/blob/master/libpod/container_inspect.go#L360-L426

Might be worth making a helper somewhere in libpod that does that job - retrieves all user-added volumes into volumes and named volumes.

Reply:
The dedup is for containers that share volumes, not necessarily for volumes one container has. the scope in `generateKubeVolumeMount` is too narrow (container level, not pod level) to have this fix the dup problem

Reply:
Ahh, makes sense.

I think we should still avoid the current method of parsing all mounts in the spec... That's asking for trouble in some ways

Reply:
an added bonus is we aren't needlessly doing an n^2 slice lookup anymore!